src/pybind/mgr/dashboard/frontend/src/app/shared/services/task-wrapper.service.ts

```diff
@@ -0,0 +1,71 @@
+import { Injectable } from '@angular/core';
+
+import { Observable } from 'rxjs/Observable';
+import { Subscriber } from 'rxjs/Subscriber';
+
+import { NotificationType } from '../enum/notification-type.enum';
+import { ExecutingTask } from '../models/executing-task';
+import { FinishedTask } from '../models/finished-task';
+import { NotificationService } from './notification.service';
+import { TaskManagerMessageService } from './task-manager-message.service';
+import { TaskManagerService } from './task-manager.service';
+
+@Injectable()
+export class TaskWrapperService {
+  constructor(
+    private notificationService: NotificationService,
+    private taskManagerMessageService: TaskManagerMessageService,
+    private taskManagerService: TaskManagerService
+  ) {}
+
+  wrapTaskAroundCall({
+    task,
+    call,
+    tasks
+  }: {
+    task: FinishedTask;
+    call: Observable<any>;
+    tasks?: ExecutingTask[];
+  }) {
+    return new Observable((observer: Subscriber<any>) => {
+      call.subscribe(
+        (resp) => {
+          if (resp.status === 202) {
+            this._handleExecutingTasks(task, tasks);
+          } else {
+            task.success = true;
+            this.notificationService.notifyTask(task);
+          }
+        },
+        (resp) => {
+          task.success = false;
+          task.exception = resp.error;
+          this.notificationService.notifyTask(task);
+          observer.error();
+        },
+        () => {
+          observer.complete();
+        }
+      );
+    });
+  }
+
+  _handleExecutingTasks(task: FinishedTask, tasks?: ExecutingTask[]) {
+    this.notificationService.show(
+      NotificationType.info,
+      task.name + ' in progress...',
```

Comment:
`task.name` is an internal `identifier` that should not be displayed to the user.

This is what you currently have implemented:

![screenshot from 2018-06-01 13-38-05](https://user-images.githubusercontent.com/14297426/40841187-77e44a36-65a1-11e8-830f-fffb4a456047.png)

Intead of `rbd/create in progress....` I would recommend a different language here, like `RBD creation in progress...` (just like we had before this refactor).

You can implement this by adding a new `string` to the `TaskManagerMessage` class, to let the developer specify the "in progress" message for each task.